contracts/task-queue/src/contract.rs

```diff
@@ -210,10 +213,25 @@ mod query {
         Ok(r)
     }
 
-    pub fn list_open(deps: Deps, env: Env) -> Result<ListOpenResponse, ContractError> {
-        // TODO: proper implementation here, this is just for minimal test code
+    // TODO: There should probably be a max page limit, but it's left unbound to keep the API simple for now
+    pub fn list_open(
+        deps: Deps,
+        env: Env,
+        start_after: Option<TaskId>,
+        limit: Option<u32>,
+    ) -> Result<ListOpenResponse, ContractError> {
+        let limit = limit.map(|v| v as usize).unwrap_or(usize::MAX);
+
         let mut open = TASKS
-            .range(deps.storage, None, None, cosmwasm_std::Order::Ascending)
+            .idx
+            .status
+            .prefix(Status::Open {}.as_str())
+            .range(
+                deps.storage,
+                start_after.map(Bound::exclusive),
+                None,
+                cosmwasm_std::Order::Ascending,
```

Comment:
I think we want descending, to start from newest first (highest id).
I mean, if we paginate and only show 20, they should be the ones I just created, not old tests.